src/meta_data.c

```diff
@@ -18,38 +22,147 @@
         return -1;                                                  \
     } while (0)
 
-static int get_serial(const struct serial_number *sn)
+static int obtain_mac_from_iface(const char *iface, uint8_t *dst_mac)
+{
+    int fd;
+    struct ifreq ifr;
+    unsigned char *mac;
+
+    fd = socket(AF_INET, SOCK_DGRAM, 0);
+    ifr.ifr_addr.sa_family = AF_INET;
+    strncpy(ifr.ifr_name , iface , IFNAMSIZ-1);
+    ioctl(fd, SIOCGIFHWADDR, &ifr);
+    close(fd);
+
+    mac = (unsigned char *)ifr.ifr_hwaddr.sa_data;
+    memcpy(dst_mac, mac, 6);
+
+	return 0;
+}
+
+/**
+ * Parse the path and obtain the PCI BDF from it
+ */
+static uint32_t get_bdf_from_path(const char *link_path)
```

Comment:
In case of WSL this algorithm will not work.  How and where the BDF number could be obtained in this case?
If we need to support WSL then need to know is it possible to get BDF number or may be just skip this interfaces? @krystian-hebel  @macpijan

Reply:
Does `lspci` work on WSL somehow?

Reply:
@0xDen It cannot be obtained because network card is not on PCI, but on a virtual bus.
This algorithm must be fixed to not crash when it encounters untypical path. For now let's ignore interfaces on which you cannot obtain BDF.

Reply:
@arturkow2000 Please fix it yourself with the proposed simplest approach and let's move on with closing that one.